### DISCUSSION TITLE
Package access modifier support for @Reducer on enums

### DISCUSSION THREAD

**Discussion by samrayner** [pointfreeco/swift-composable-architecture#2936] [Ideas]
Didn't want to raise this as a bug, as it's not at all urgent, but `public` is applied correctly when using @Reducer on a `public` enum, however nested types inside a `package` enum do not inherit the `package` access modifier (so default to `internal`). For now I'm happy to just use `public` for the enum but it would be nice to scope it more accurately.

I think it probably just requires a tweak to https://github.com/pointfreeco/swift-composable-architecture/blob/49dae77fc89445bb4aa909a636d6e415af0e1343/Sources/ComposableArchitectureMacros/ReducerMacro.swift#L193 but I'm not familiar enough with SwiftSyntax I'm afraid!

Thanks!

**Comment by mbrandonw** [comment]
Hi @samrayner, it does seem like an oversight of the `@Reducer` macro when it comes to `package`. Are you interested in PR'ing a fix? If not we will get to in the coming days.

**Reply by samrayner** [reply]
Gave it a go https://github.com/pointfreeco/swift-composable-architecture/pull/2939
